warp10/src/main/java/io/warp10/script/functions/PUT.java

```diff
@@ -25,30 +25,30 @@
 import java.util.Map;
 
 /**
- * Pushes a value into a map. Modifies the map on the stack.
+ * Pushes a value into a map or list. Modifies the map or list on the stack.
  */
 public class PUT extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
+
   public PUT(String name) {
     super(name);
   }
-  
+
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
-    
+
     Object key = stack.pop();
-    Object value = stack.pop();    
-    
+    Object value = stack.pop();
+
     Object maporlist = stack.peek();
 
     if (maporlist instanceof Map) {
-      ((Map) maporlist).put(key, value);      
+      ((Map) maporlist).put(key, value);
     } else if (maporlist instanceof List) {
       if (!(key instanceof Number)) {
```

Comment:
Here too the key should be checked against Long instead of Number as Doubles will pass through